### PR TITLE
chore: Drop conversion webhooks

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
Dropping conversion webhook.

**How was this change tested?**
Manually tested

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: kwok/charts/crds/karpenter.sh_nodeclaims.yaml [hunk 1]
```diff
@@ -816,14 +816,3 @@ spec:
       storage: false
       subresources:
         status: {}
-  conversion:
```
Comment (engedaam) [inline kwok/charts/crds/karpenter.sh_nodeclaims.yaml:819]: We need to drop the automation here: https://github.com/kubernetes-sigs/karpenter/blob/main/hack/kwok/conversion.sh

Comment (engedaam) [inline kwok/charts/crds/karpenter.sh_nodeclaims.yaml:819]: This also means we need to edit the Makefile

File: pkg/operator/injection/injection.go [hunk 2]
```diff
@@ -72,12 +69,3 @@ func GetClient(ctx context.Context) client.Client {
 	c := ctx.Value(clientKey)
 	return c.(client.Client)
 }
-
-func WithNodeClasses(ctx context.Context, opts []schema.GroupVersionKind) context.Context {
```
Comment (engedaam) [inline pkg/operator/injection/injection.go:76]: We should also be dropping the context injection for the client

File: go.mod [hunk 3]
```diff
@@ -34,13 +33,15 @@ require (
 	sigs.k8s.io/controller-runtime v0.18.5
 )
 
-require github.com/klauspost/compress v1.17.9 // indirect
+require (
+	github.com/klauspost/compress v1.17.9 // indirect
```
Comment (engedaam) [inline go.mod:37]: Are we adding a dependacey here?

Comment (engedaam) [reply go.mod:37]: Also can we drop the knative dependacey in this PR?

File: pkg/operator/operator.go [hunk 4]
```diff
@@ -241,36 +222,17 @@ func (o *Operator) WithControllers(ctx context.Context, controllers ...controlle
...
-			})
-			ctx = injection.WithNodeClasses(ctx, gvk)
-			ctx = injection.WithClient(ctx, o.GetClient())
-			webhooks.Start(ctx, o.GetConfig(), o.webhooks...)
-		}()
-	}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ctx = injection.WithClient(ctx, o.GetClient())
```
Comment (engedaam) [inline pkg/operator/operator.go:235]: We should not need to inject the client into the context. This was only needed for the expireAfter conversion

File: pkg/operator/operator.go [hunk 5]
```diff
@@ -241,36 +222,17 @@ func (o *Operator) WithControllers(ctx context.Context, controllers ...controlle
...
-			// Taking the first supported NodeClass to be the default NodeClass
-			gvk := lo.Map(cp.GetSupportedNodeClasses(), func(nc status.Object, _ int) schema.GroupVersionKind {
-				return object.GVK(nc)
-			})
-			ctx = injection.WithNodeClasses(ctx, gvk)
-			ctx = injection.WithClient(ctx, o.GetClient())
-			webhooks.Start(ctx, o.GetConfig(), o.webhooks...)
-		}()
-	}
+	wg.Add(1)
```
Comment (engedaam) [inline pkg/operator/operator.go:232]: Why do we still need this block?

**Review by engedaam — COMMENTED** [review]
Had a few comments

File: kwok/charts/crds/karpenter.sh_nodeclaims.yaml (hunk 1 above)
Comment (jigisha620) [reply kwok/charts/crds/karpenter.sh_nodeclaims.yaml:819]: Yes right, dropped.

File: pkg/operator/injection/injection.go [hunk 6]
```diff
@@ -64,20 +61,7 @@ func WithOptionsOrDie(ctx context.Context, opts ...options.Injectable) context.C
 	return ctx
 }
 
-func WithClient(ctx context.Context, client client.Client) context.Context {
-	return context.WithValue(ctx, clientKey, client)
-}
-
 func GetClient(ctx context.Context) client.Client {
```
Comment (engedaam) [inline pkg/operator/injection/injection.go:64]: We should also drop `GetClient

File: pkg/operator/injection/injection.go [hunk 7]
```diff
@@ -22,19 +22,16 @@ import (
...
 
 	"github.com/samber/lo"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"sigs.k8s.io/karpenter/pkg/operator/options"
 )
 
 type controllerNameKeyType struct{}
 type clientKeyType struct{}
```
Comment (engedaam) [inline pkg/operator/injection/injection.go:31]: We can also drop the client key struct

**Review by engedaam — COMMENTED** [review]
Two comments

**Review by engedaam — COMMENTED** [review]
/lgtm

File: kwok/main.go [hunk 8]
```diff
@@ -34,12 +33,11 @@ func main() {
...
 	cloudProvider := kwok.NewCloudProvider(ctx, op.GetClient(), instanceTypes)
 	op.
-		WithWebhooks(ctx, webhooks.NewWebhooks()...).
 		WithControllers(ctx, controllers.NewControllers(
 			op.Manager,
 			op.Clock,
 			op.GetClient(),
 			op.EventRecorder,
 			cloudProvider,
-		)...).Start(ctx, cloudProvider)
```
Comment (jonathan-innis) [inline kwok/main.go:44]: Note: This is another signal that this change is breaking for this upstream library

File: pkg/operator/operator.go [hunk 9]
```diff
@@ -125,14 +115,6 @@ func NewOperator() (context.Context, *Operator) {
 		debug.SetMemoryLimit(newLimit)
 	}
 
-	// Webhook
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:128]: Do we need L116 anymore if we don't use knative anymore? Maybe this gets taken care of in a different PR to fully remove?

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**